src/kokoro_model.h

```diff
@@ -0,0 +1,456 @@
+	struct ggml_tensor * window;
+
+	struct ggml_tensor * m_source_weight;
+	struct ggml_tensor * m_source_bias;
+	struct ggml_tensor * out_conv_weight;
+	struct ggml_tensor * out_conv_bias;
+	std::vector<kokoro_noise_residual_block*> noise_blocks;
+	std::vector<kokoro_generator_residual_block*> res_blocks;
+	std::vector<kokoro_generator_upsample_block*> ups;
+};
+
+struct kokoro_decoder {
+	struct ggml_tensor * f0_conv;
+	struct ggml_tensor * f0_conv_bias;
+	struct ggml_tensor * n_conv;
+	struct ggml_tensor * n_conv_bias;
+	struct ggml_tensor * asr_conv;
+	struct ggml_tensor * asr_conv_bias;
+	std::vector<ada_residual_conv_block*> decoder_blocks;
+	ada_residual_conv_block* encoder_block;
+	kokoro_generator * generator;
+};
+
+struct albert_layer {
+	struct ggml_tensor * ffn;
+	struct ggml_tensor * ffn_out;
+	struct ggml_tensor * ffn_bias;
+	struct ggml_tensor * ffn_out_bias;
+	struct ggml_tensor * layer_output_norm_weight;
+	struct ggml_tensor * layer_output_norm_bias;
+	struct ggml_tensor * q;
+	struct ggml_tensor * k;
+	struct ggml_tensor * v;
+	struct ggml_tensor * o;
+	struct ggml_tensor * q_bias;
+	struct ggml_tensor * k_bias;
+	struct ggml_tensor * v_bias;
+	struct ggml_tensor * o_bias;
+	struct ggml_tensor * attn_norm_weight;
+	struct ggml_tensor * attn_norm_bias;
+};
+
+struct kokoro_model : tts_model {
+	// standard configruation for Kokoro's Albert model
+	// tokenization
+	uint32_t bos_token_id = 0;
+	uint32_t eos_token_id = 0;
+	// duration prediction
+	uint32_t max_context_length = 512;
+	uint32_t vocab_size = 178;
+	uint32_t hidden_size = 768;
+	uint32_t n_attn_heads = 12;
+	uint32_t n_layers = 1;
+	uint32_t n_recurrence = 12;
+	uint32_t head_size = 64;
+	uint32_t duration_hidden_size = 512;
+	//uint32_t duration_sample_upsample_factor;
+	uint32_t up_sampling_factor;
+	float upsample_scale = 300.0f;
+	float scale = 0.125f;
+
+	// standard configuration for duration prediction
+	uint32_t f0_n_blocks = 3;
+	uint32_t n_duration_prediction_layers = 3;
+	// while it is technically possible for the duration predictor to assign 50 values per token there is no practical need to 
+	// allocate that many items to the sequence as it is impossible for all tokens to require such long durations and each 
+	// allocation increases node allocation size by O(N)
+	uint32_t max_duration_per_token = 20;
+	uint32_t style_half_size = 128;
+
+	// standard text encoding configuration
+	uint32_t n_conv_layers = 3;
+
+	// standard decoder configuration
+	uint32_t n_kernels = 3;
+	uint32_t n_upsamples = 2;
+	uint32_t n_decoder_blocks = 4;
+	uint32_t n_res_blocks = 6;
+	uint32_t n_noise_blocks = 2;
+	uint32_t out_conv_padding = 3;
+	uint32_t post_n_fft = 11;
+	uint32_t true_n_fft = 20;
+	uint32_t stft_hop = 5;
+	uint32_t harmonic_num = 8;
+	float sin_amp = 0.1f;
+	float noise_std = 0.003f;
+	float voice_threshold = 10.0f;
+	float sample_rate = 24000.0f;
+	std::string window = "hann"; 
```

Comment:
Some of this configuration is not configured via the gguf file. Configuration therein should be supported.